test/apps/iodemo/io_demo.cc

```diff
@@ -1158,8 +1174,9 @@ class DemoClient : public P2pDemoCommon {
 
         while ((total_iter < opts().iter_count) && (_status == OK)) {
             VERBOSE_LOG << " <<<< iteration " << total_iter << " >>>>";
-
-            wait_for_responses(opts().window_size - 1);
+            wait_for_responses(std::min(opts().window_size - 1,
+                                        opts().conn_window_size *
+                                        (long)_active_servers.size()));
```

Comment:
as evgeny said:
```
max_outstanding = std::min(opts().window_size, 
                           opts().conn_window_size * _active_servers.size())
wait_for_responses(max_outstanding - 1)
```

Reply:
@yosefe if the size of the _active_servers is 0, at the beginning for example, then it's wait_for_responses(-1).
why is the -1 needed in (max_outstanding - 1) ?

Reply:
maybe move wait_for_responses to after checking _active_servers.empty()

Reply:
to before connect_all()? isn't it better to progress first?

but why is the way it was not ok? without -1

Reply:
without -1 it will not work: it will wait until no more than window_size are outstanding. and then we will send 1 more so there will actually be (window_size+1) outstanding.

Reply:
even if we do:
wait_for_responses(std::min(opts().window_size **- 1**,  opts().conn_window_size *(long)_active_servers.size()));
?

Reply:
in this case this function can return when no server is capable of sending